src/omv/modules/py_tf.c

```diff
@@ -619,12 +575,252 @@ STATIC mp_obj_t py_tf_segment(uint n_args, const mp_obj_t *pos_args, mp_map_t *k
+            list_lnk_t *old_it = it;
+            it = it->next;
+
+            float iou = py_tf_yolo_iou(&lnk_data, lnk_data2);
+            // Do not use fast_expf() as it does not output 1 when it's input is 0.
+            // This will cause the scores of non-overlapping bounding boxes to decay.
+            lnk_data2->score *= expf(sigma_scale * iou * iou);
+
+            if (lnk_data2->score < arg->nms_threshold) {
+                list_remove(&detections, old_it, NULL);
+            } else if (lnk_data2->score > max_score) {
+                max_score = lnk_data2->score;
+                max_it = old_it;
+            }
+        }
+
+        // Find the maximum label index for the output list.
+        max_label = IM_MAX(lnk_data.label_index, max_label);
+    }
+
+    // Create a list per class label.
+    *(arg->output) = mp_obj_new_list(max_label + 1, NULL);
+    for (size_t i = 0; i <= max_label; i++) {
+        ((mp_obj_list_t *) *(arg->output))->items[i] = mp_obj_new_list(0, NULL);
+    }
+
+    float x_scale = arg->roi->w / ((float) params->input_width);
+    float y_scale = arg->roi->h / ((float) params->input_height);
+    // MAX == KeepAspectRatioByExpanding - MIN == KeepAspectRatio
+    float scale = IM_MIN(x_scale, y_scale);
+    int x_offset = fast_floorf((arg->roi->w - (params->input_width * scale)) / 2.0f) + arg->roi->x;
+    int y_offset = fast_floorf((arg->roi->h - (params->input_height * scale)) / 2.0f) + arg->roi->y;
+
+    size_t len = list_size(&nms_detections);
+    for (size_t i = 0; i < len; i++) {
+        py_tf_yolo_output_callback_lnk_data_t lnk_data;
+        list_pop_front(&nms_detections, &lnk_data);
+        py_tf_classification_obj_t *o = m_new_obj(py_tf_classification_obj_t);
+        o->base.type = &py_tf_classification_type;
+        o->x = mp_obj_new_int(fast_floorf(lnk_data.rect.x * scale) + x_offset);
+        o->y = mp_obj_new_int(fast_floorf(lnk_data.rect.y * scale) + y_offset);
+        o->w = mp_obj_new_int(fast_floorf(lnk_data.rect.w * scale));
+        o->h = mp_obj_new_int(fast_floorf(lnk_data.rect.h * scale));
+        o->output = mp_obj_new_float(lnk_data.score);
+        mp_obj_list_append(((mp_obj_list_t *) *(arg->output))->items[lnk_data.label_index], o);
+    }
+}
+#endif // IMLIB_ENABLE_TF_YOLO
+
 STATIC mp_obj_t py_tf_detect(uint n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
-    enum { ARG_roi, ARG_thresholds, ARG_invert };
+    enum {
+        ARG_roi, ARG_thresholds, ARG_invert, ARG_model, ARG_score_threshold, ARG_label_threshold,
+        ARG_nms_threshold, ARG_nms_sigma
+    };
     static const mp_arg_t allowed_args[] = {
         { MP_QSTR_roi, MP_ARG_OBJ | MP_ARG_KW_ONLY, {.u_rom_obj = MP_ROM_NONE} },
         { MP_QSTR_thresholds, MP_ARG_OBJ | MP_ARG_KW_ONLY, {.u_rom_obj = MP_ROM_NONE} },
         { MP_QSTR_invert,  MP_ARG_INT | MP_ARG_KW_ONLY, {.u_bool = false } },
+        { MP_QSTR_model,  MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = PY_TF_MODEL_FOMO } },
```

Comment:
```suggestion
        { MP_QSTR_model,  MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = -1 } },
```
For auto-detection of model type.

Reply:
Done

Reply:
I added the signature support. However, I strongly recommend we leave this option as PY_TF_MODEL_FOMO right now and collect many model signatures before enabling the feature. This will prevent FOMO models from being broken. I'll ask edge impulse if they can provide a few models to collect more signatures from. However, even for the NVIDIA TAO V3 models using INT8 I found 4 unique signatures...

Reply:
> However, even for the NVIDIA TAO V3 models using INT8 I found 4 unique signatures...
Post the models ?

Reply:
It's 4 out of the 7 I sent you.